woocommerce/woocommerce
# Fix: Upsells collection not showing correct products on frontend

### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/pull/51730

As reported on issue #51601, If a specific product is selected on Single Product Template then Upsells collection isn't showing correct products on frontend. I found out that there is a typo in variable name which is causing the issue.

**Note:** This PR fix the issue for frontend. On Editor, it will still show random products even if specific product is selected. As this isn't super urgent, we can address it in follow-up PR. It's important that shopper facing pages are fixed.



### How to test the changes in this Pull Request:



1. Go to Single Product Template 
2. Add Product Collection block and choose "Upsells" collection
3. Select "From a specific product" in Inspector control and select specific product which have Upsells products
![image](https://github.com/user-attachments/assets/cf4e6ec4-947f-4ad7-9984-9a90c8efa6ca)

5. Publish the post and go to frontend
6. Verify that Upsells products are displayed correctly for the selected product
	- **Note:** If you don't have any product which have Upsells products, you will need to create one. OR add upsells products to existing product.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 
Product Collection: Fix Upsells collection not showing linked product on Frontend on Single Product Template
</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
